### PR TITLE
Add ng-swagger-gen to automatically generate models and services with swagger.yaml

### PR DESCRIPTION
**Pull request by AllForNothing**
To automatically generate models and services with swagger.yaml

Signed-off-by: sshijun <sshijun@vmware.com>

### CODE REVIEW

**Comment by jwangyangls** [comment]
I think such a modification greatly facilitates the development of the front end; the disadvantage is that ng-swagger-gen cannot support the yaml format file for the time being.

**Comment by jwangyangls** [comment]
Automatically generated files I recommend against uploading to GitHub

File: src/portal/package.json [hunk 1]
```diff
@@ -16,7 +16,11 @@
     "pree2e": "webdriver-manager update",
     "e2e": "protractor",
     "build": "ng build --aot",
-    "release": "ng build --prod"
+    "release": "ng build --prod",
+    "generate_swagger_json": "node scripts\\convert-yaml-to-json.js",
```
Comment (jwangyangls) [inline src/portal/package.json]: '\\'  maybe not work in some environments

Comment (AllForNothing) [reply src/portal/package.json]: > “\” “\” maybe not work in some environments

Fixed

**Comment by heww** [comment]
Generated go files from swagger.yaml are excluded in git, I think generated typescript files also need to exclude in git.

**Comment by heww** [comment]
And for legacy APIs we will not update them as little as possible so generate models and client for legacy APIs is not cost effective.

**Comment by AllForNothing** [comment]
> Please exclude generated typescript files in git and not generate files for legacy APIs.

If excluding generated typescript files in git , we need to modify all the related CI for building and running.

@zhoumeina Please help to make a decision

**Pull request merged**